### PR TITLE
fix(profiling): fix deprecated ddtrace usage

### LINKED ISSUE #8881: Importing profiler causes deprecation warning that can fail tests
### Summary of problem

Running `pytest -W "error::ddtrace.DDTraceDeprecationWarning" tests` with a test that imports `from ddtrace.profiling.profiler import Profiler` causes https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/profiling/event.py#L6 to be hit, but as of 2.7.0, `ddtrace.span` is now `ddtrace._tracer.span` and this fires a deprecation warning which fails the test.

### Which version of dd-trace-py are you using?

>=2.7.0

### Which version of pip are you using?

22.3.1

### How can we reproduce your problem?

test.py:
```
class TestProfiler:
    def test_profiling(self) -> None:
        from ddtrace.profiling.profiler import Profiler
        datadog_profiler = Profiler()
        assert datadog_profiler is not None
```

Run with

`pytest -W "error::ddtrace.DDTraceDeprecationWarning" test.py`

### What is the result that you get?

```
FAILED  test.py::TestProfiler::test_profiling - ddtrace.DDTraceDeprecationWarning: The span module is deprecated and will be moved.: A new span interface will be provided by the trace sub-package.
```

### What is the result that you expected?

Test passes.

### PR DESCRIPTION
**Pull request by sanchda**
As reported in #8881, the profiler incorrectly imported the deprecated span.

Fixes #8881 

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/profiling/test_profiler.py [hunk 1]
```diff
@@ -411,3 +411,15 @@ def test_profiler_serverless(monkeypatch):
...
+@pytest.mark.subprocess()
+def test_profiler_ddtrace_deprecation():
+    """
+    ddtrace interfaces loaded by the profiler can be marked deprecated, and we should update
+    them wen this happens.  As reported by https://github.com/DataDog/dd-trace-py/issues/8881
+    """
+    import warnings
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", DeprecationWarning)
```
Comment (brettlangdon) [inline tests/profiling/test_profiler.py:425]: do we need to do something in this block?

I think `catch_warnings` only works for code executed within the block, so we might need to import the profiler or do something here?

Comment (sanchda) [reply tests/profiling/test_profiler.py:425]: Haha, linter was helpfully removing unused imports.

File: releasenotes/notes/profiling-fix-deprecated-span-usage-512723136f1682d2.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fixes a defect where the deprecated path to the Datadog span type was used
```
Comment (emmettbutler) [inline releasenotes/notes/profiling-fix-deprecated-span-usage-512723136f1682d2.yaml:4]:
```suggestion
    profiling: Fixes a defect where the deprecated path to the Datadog span type was used
```

**Review by emmettbutler — APPROVED** [review]

File: releasenotes/notes/profiling-fix-deprecated-span-usage-512723136f1682d2.yaml (hunk 2 above)
Comment (sanchda) [reply releasenotes/notes/profiling-fix-deprecated-span-usage-512723136f1682d2.yaml]: TYVM!

**Pull request merged**